src/applications/simple-forms/21-0966/config/form.js

```diff
@@ -128,10 +185,94 @@ const formConfig = {
       pages: {
         veteranBenefitSelection: {
           path: 'veteran-benefit-selection',
-          depends: formData => preparerIsVeteran({ formData }),
+          depends: formData =>
+            preparerIsVeteran({ formData }) &&
+            noActiveITFOrCreationFailed({ formData }),
           title: formData => benefitSelectionChapterTitle({ formData }),
           uiSchema: veteranBenefitSelection.uiSchema,
           schema: veteranBenefitSelection.schema,
+          onNavForward: ({ formData, goPath, goNextPath, setFormData }) => {
+            const benefitTypesArray = [];
+
+            if (formData.benefitSelection?.compensation) {
+              benefitTypesArray.push('compensation');
+            }
+            if (formData.benefitSelection?.pension) {
+              benefitTypesArray.push('pensionSurvivorsBenefits');
+            }
+
+            goPath('get-itf-status');
+
+            Promise.all(
+              benefitTypesArray.map(benefitType =>
+                apiRequest(`/intent_to_file/${benefitType}`, {
+                  method: 'POST',
+                }),
+              ),
+            )
+              .then(() => {
+                goPath('confirmation');
+              })
+              .catch(() => {
+                setFormData({ ...formData, itfCreationFailed: true });
+
+                goNextPath();
+              });
+          },
+        },
+        veteranBenefitSelectionCompensation: {
+          path: 'veteran-benefit-selection-compensation',
+          depends: formData =>
+            preparerIsVeteran({ formData }) &&
+            hasActivePensionITF({ formData }),
+          title: formData => benefitSelectionChapterTitle({ formData }),
+          uiSchema: veteranBenefitSelectionCompensation.uiSchema,
+          schema: veteranBenefitSelectionCompensation.schema,
+          onNavForward: ({ formData, goPath, setFormData }) => {
+            if (formData.benefitSelectionCompensation) {
+              goPath('get-itf-status');
+
+              apiRequest('/intent_to_file/compensation', { method: 'POST' })
+                .then(() => {
+                  goPath('confirmation');
```

Comment:
Do we need to set the expiration date or anything that gets returned or did we settle on just displaying one year from "now" and ignoring any response from the API?

Reply:
That's a good point. I could probably grab the data from the response here and set it to something to display on the confirmation page. I'll do that when I loop back through all of this and fix the UI